ddtrace/contrib/botocore/patch.py

```diff
@@ -183,11 +182,12 @@ def prep_context_injection(ctx, endpoint_name, operation, trace_operation, param
     if endpoint_name == "events" and operation == "PutEvents":
         injection_function = inject_trace_to_eventbridge_detail
         cloud_service = "events"
-    if endpoint_name == "sns" and operation == "Publish":
-        injection_function = inject_trace_to_sqs_or_sns_message
-        cloud_service = "sns"
-    if endpoint_name == "sns" and operation == "PublishBatch":
-        injection_function = inject_trace_to_sqs_or_sns_batch_message
+    if endpoint_name == "sns" and "Publish" in operation:
+        injection_function = (  # noqa: E731
+            lambda ctx, params, span, endpoint_service: inject_trace_to_sqs_or_sns_message(
```

Comment:
just so I understand, this is just so we are dropping the `span` attribute?

this is bike shedding a bit, but it might be more performant to not use a lambda here, e.g.:


```python
def inject_trace_to_sqs_or_sns_message(ctx, params, span, endpoint_service):
    return update_messages(ctx, params, endpoint_service)


....

injection_function = inject_trace_to_sqs_or_sns_message
```